src/Utils/Types/VendorState.php

```diff
@@ -4,8 +4,8 @@
 
 class VendorState
 {
-    const INSERT = 'insert';
-    const UPDATE = 'update';
-    const DELETE = 'delete';
-    const DELETE_AND_UPDATE = 'deleteAndUpdate';
+    final public const INSERT = 'insert';
+    final public const UPDATE = 'update';
+    final public const DELETE = 'delete';
+    final public const DELETE_AND_UPDATE = 'deleteAndUpdate';
```

Comment:
We should look into converting thes to enums (later both here and in imports :-))